packages/salesforcedx-vscode-core/test/jest/services/sdr/componentSetUtils.test.ts

```diff
@@ -4,61 +4,187 @@
 
-      const dummyComponentSet = new ComponentSet();
-      await setApiVersion(dummyComponentSet);
+      const getUserConfiguredApiVersionMock = jest.fn().mockResolvedValue(undefined);
+      ConfigUtil.getUserConfiguredApiVersion = getUserConfiguredApiVersionMock;
 
+      const getOrgApiVersionMock = jest
+        .spyOn(componentSetUtils, 'getOrgApiVersion')
+        .mockResolvedValue('');
+
+      await componentSetUtils.setApiVersion(componentSet);
+
+      expect(componentSet.apiVersion).toBe('44.0');
       expect(getUserConfiguredApiVersionMock).toHaveBeenCalled();
-      expect(workspaceContextGetConnectionMock).toHaveBeenCalled();
-      expect(dummyComponentSet.apiVersion).toEqual(orgApiVersion);
+      expect(getOrgApiVersionMock).toHaveBeenCalled();
+    });
+  });
+
+  describe('setSourceApiVersion', () => {
+    it('should validate that sourceApiVersion is set from the ComponentSet (metadata file) when present', async () => {
+      // *** Set (faked) componentSet.sourceApiVersion
+      const componentSet = new ComponentSet();
+      componentSet.sourceApiVersion = '54.0';
+
+      const getValueMock = jest.fn().mockResolvedValue('53.0');
+      SfdxProjectConfig.getValue = getValueMock;
+
+      const getUserConfiguredApiVersionMock = jest.fn().mockResolvedValue('52.0');
+      ConfigUtil.getUserConfiguredApiVersion = getUserConfiguredApiVersionMock;
+
+      const getOrgApiVersionMock = jest
+        .spyOn(componentSetUtils, 'getOrgApiVersion')
+        .mockResolvedValue('51.0');
+
+      await componentSetUtils.setSourceApiVersion(componentSet);
+
+      expect(componentSet.sourceApiVersion).toBe('54.0');
+      expect(getValueMock).not.toHaveBeenCalled();
+      expect(getUserConfiguredApiVersionMock).not.toHaveBeenCalled();
+      expect(getOrgApiVersionMock).not.toHaveBeenCalled();
+    });
+
+    it('should validate that sourceApiVersion is set from SfdxProjectConfig when not set via a ComponentSet (metadata file)', async () => {
+      // *** set/fake componentSet.sourceApiVersion
+      const componentSet = new ComponentSet();
+      componentSet.sourceApiVersion = undefined;
+
+      const getValueMock = jest.fn().mockResolvedValue('53.0');
+      SfdxProjectConfig.getValue = getValueMock;
+
+      const getUserConfiguredApiVersionMock = jest.fn().mockResolvedValue(undefined);
+      ConfigUtil.getUserConfiguredApiVersion = getUserConfiguredApiVersionMock;
+
+      const getOrgApiVersionMock = jest
+        .spyOn(componentSetUtils, 'getOrgApiVersion')
+        .mockResolvedValue('51.0');
+
+      await componentSetUtils.setSourceApiVersion(componentSet);
+
+      expect(componentSet.sourceApiVersion).toBe('53.0');
+      expect(getValueMock).toHaveBeenCalled();
+      expect(getUserConfiguredApiVersionMock).not.toHaveBeenCalled();
+      expect(getOrgApiVersionMock).not.toHaveBeenCalled();
+    });
+
+    it('should validate that sourceApiVersion is set from getUserConfiguredApiVersion', async () => {
```

Comment:
May be just add 'when not set (undefined) through componentSet or SfdxProjectConfig'

Reply:
done